dcapp/app/src/main/java/com/example/michael/mqtttest/BroadcastService.java

```diff
@@ -77,8 +93,75 @@
     private static final int Y_ACCELERATION_INDEX = 1;
     private static final int Z_ACCELERATION_INDEX = 2;
 
-    private float[] gravity = { 0 , 0 , 0};
-    private final float alpha = (float) 0.8;
+    private static final int FASTEST_PUBLICATION_RATE = 200;
+
+    private ScheduledFuture publicationHandle;
+    private final Runnable publishData = new Runnable() {
+        private boolean isConnecting = false;
```

Comment:
This is probably not thread safe but it should be fine?  might consider using the `synchronized` keyword